src/Symmetry/LocalOperators.jl

```diff
@@ -0,0 +1,393 @@
+    dE_dz +=    (c2[2]*real(Jp¹)+c2[4]*imag(Jp¹))dA_dz[2] +
+                c2[3]*dA_dz[1]
+    dE_dJp +=   (2/2)*(c2[1]*Jp¹-im*c2[5]*Jp¹)A[3] +
+                (1/2)*(c2[2]    -im*c2[4]    )A[2]
+
+    max_k == 2 && @goto exit
+
+    # Quartic contributions
+
+    X² = X*X
+    Jp³ = Jp²*Jp¹
+    Jz³ = Jz²*Jz¹
+    Jp⁴ = Jp²*Jp²
+    Jz⁴ = Jz²*Jz²
+
+    A = (35Jz⁴ - (30X)Jz² + (3X²),
+        7Jz³ - (3X)Jz¹,
+        7Jz² - (X),
+        Jz¹,
+        1)
+    dA_dz = (140Jz³ - (60X)Jz¹,
+            21Jz² - 3X,
+            14Jz¹,
+            1)
+    E +=        (c4[1]*real(Jp⁴)+c4[9]*imag(Jp⁴))A[5] +
+                (c4[2]*real(Jp³)+c4[8]*imag(Jp³))A[4] +
+                (c4[3]*real(Jp²)+c4[7]*imag(Jp²))A[3] +
+                (c4[4]*real(Jp¹)+c4[6]*imag(Jp¹))A[2] +
+                c4[5]*A[1]
+    dE_dz +=    (c4[2]*real(Jp³)+c4[8]*imag(Jp³))dA_dz[4] +
+                (c4[3]*real(Jp²)+c4[7]*imag(Jp²))dA_dz[3] +
+                (c4[4]*real(Jp¹)+c4[6]*imag(Jp¹))dA_dz[2] +
+                c4[5]*dA_dz[1]
+    dE_dJp +=   (4/2)*(c4[1]*Jp³-im*c4[9]*Jp³)A[5] +
+                (3/2)*(c4[2]*Jp²-im*c4[8]*Jp²)A[4] +
+                (2/2)*(c4[3]*Jp¹-im*c4[7]*Jp¹)A[3] +
+                (1/2)*(c4[4]    -im*c4[6]    )A[2]
+
+    max_k == 4 && @goto exit
+
+    # Hexic contributions
+
+    X³ = X²*X
+    Jp⁵ = Jp⁴*Jp¹
+    Jz⁵ = Jz⁴*Jz¹
+    Jp⁶ = Jp³*Jp³
+    Jz⁶ = Jz³*Jz³
+
+    A = (231Jz⁶ - (315X)Jz⁴ + (105X²)Jz² - (5X³),
+        33Jz⁵ - (30X)Jz³ + (5X²)Jz¹,
+        33Jz⁴ - (18X)Jz² + (X²),
+        11Jz³ - (3X)Jz¹,
+        11Jz² - (X),
+        Jz¹,
+        1)
+    dA_dz = (1386Jz⁵ - (1260X)Jz³ + (210X²)Jz¹,
+            165Jz⁴ - (90X)Jz² + 5X²,
+            132Jz³ - (36X)Jz¹,
+            33Jz² - 3X,
+            22Jz¹,
+            1)
+    E +=        (c6[1]*real(Jp⁶)+c6[13]*imag(Jp⁶))A[7] +
+                (c6[2]*real(Jp⁵)+c6[12]*imag(Jp⁵))A[6] +
+                (c6[3]*real(Jp⁴)+c6[11]*imag(Jp⁴))A[5] +
+                (c6[4]*real(Jp³)+c6[10]*imag(Jp³))A[4] +
+                (c6[5]*real(Jp²)+c6[9] *imag(Jp²))A[3] +
+                (c6[6]*real(Jp¹)+c6[8] *imag(Jp¹))A[2] +
+                c6[7]*A[1]
+    dE_dz +=    (c6[2]*real(Jp⁵)+c6[12]*imag(Jp⁵))dA_dz[6] +
+                (c6[3]*real(Jp⁴)+c6[11]*imag(Jp⁴))dA_dz[5] +
+                (c6[4]*real(Jp³)+c6[10]*imag(Jp³))dA_dz[4] +
+                (c6[5]*real(Jp²)+c6[9] *imag(Jp²))dA_dz[3] +
+                (c6[6]*real(Jp¹)+c6[8] *imag(Jp¹))dA_dz[2] +
+                c6[7]*dA_dz[1]
+    dE_dJp +=   (6/2)*(c6[1]*Jp⁵-im*c6[13]*Jp⁵)A[7] +
+                (5/2)*(c6[2]*Jp⁴-im*c6[12]*Jp⁴)A[6] +
+                (4/2)*(c6[3]*Jp³-im*c6[11]*Jp³)A[5] +
+                (3/2)*(c6[4]*Jp²-im*c6[10]*Jp²)A[4] +
+                (2/2)*(c6[5]*Jp¹-im*c6[9] *Jp¹)A[3] +
+                (1/2)*(c6[6]    -im*c6[8]     )A[2]
+
+    # Unpack gradient components
+
+    @label exit
+    dE_dx = +2real(dE_dJp)
+    dE_dy = -2imag(dE_dJp)
+    return (E, Vec3(dE_dx, dE_dy, dE_dz))
+end
```

Comment:
I have to admit that I think I would need a while to fully unpack everything here. But it's clearly very well written.